test/api/test_default_schema.cpp

```diff
@@ -0,0 +1,22 @@
+#include "catch.hpp"
+#include "test_helpers.hpp"
+
+using namespace duckdb;
+using namespace std;
+
+TEST_CASE("Test Default Schema", "[api]") {
+	unique_ptr<QueryResult> result;
+	DuckDB db(nullptr);
+	Connection con(db);
+
+	REQUIRE_NO_FAIL(con.Query("CREATE SCHEMA s1;"));
+	REQUIRE_NO_FAIL(con.Query("CREATE TABLE s1.t1(a INTEGER, b VARCHAR);"));
```

Comment:
I would like to see a more tests:
* What happens if the schema does not exist?
* What if we set an empty schema? (`PRAGMA default_schema=''`)
* What if we set the schema to NULL? (`PRAGMA default_schema=NULL`)
* Can we still use the `main` schema explicitly after changing the default schema, e.g. what if we have a table in the main schema, can we still use `main.t1` to access it as expected even after we have changed the default schema?
* What about other catalog entries (views, sequences, macros)?
* What about temporary tables, do they still work when you change the default schema?
* What about other types of statements that might be affected by this change (function binding, aggregates, alter/drop table statements, deletes, updates)?
* Specifically functions I'm worried about, since all functions currently live in the default "main" schema setting the default schema might require you to do e.g. `select main.sqrt(i) from table)` which I think it undesirable. Could you test this? If this does in fact happen, perhaps we should think about ways of preventing this from happening (e.g. functions always go to the main schema).

Reply:
Thanks for the pointers and suggestions! Answers in the order you asked:

- The pragma fails.
- The pragma fails.
- The pragma fails.
- Yes, as one would expect.
- All of those work as expected.
- Temporary tables work as expected.
- All of those statements work as expected.
- Yes, I verified that what you suspected is correct, i.e., one must provide the `main` schema name explicitly when referring to (even built-in) functions. I agree, this is quite tedious. Putting all functions in the main schema might make sense. Are there any use-cases you can think of when a user would want to create functions within a (non-main) schema?

I've added tests for all of the points above except functions.